crud.c

```diff
@@ -0,0 +1,222 @@
+    	printf("Invalid Age, Please try again \n Enter Age : ");
+    	scanf(" %d", &user.age);
+	}
+    fprintf(file, "%s %s %d\n", user.id, user.name, user.age);
+    fclose(file);
+    printf("User added successfully!\n");
+}
+
+void listUsers() {
+    User user;
+    FILE *file = fopen(FILENAME, "r");
+
+    if (file == NULL) {
+        printf("Error: Could not open file for reading.\n");
+        return;
+    }
+
+    printf("\nUsers\n");
+    while (fscanf(file, "%s %s %d", &user.id, user.name, &user.age) == 3) {
+        printf("ID: %s, Name: %s, Age: %d\n", user.id, user.name, user.age);
+    }
+
+    fclose(file);
+}
+
+void modifyUser() {
+    char idToUpdate[15], found = 0;
+    User user;
+    FILE *file = fopen(FILENAME, "r");
+    FILE *tempFile = fopen("temp.txt", "w");
+
+    if (file == NULL || tempFile == NULL) {
+        printf("Error: Could not open file for reading or writing.\n");
+        return;
+    }
+
+    printf("Enter User ID to modify: ");
+    scanf("%s", idToUpdate);
+
+    while (fscanf(file, "%s %s %d", user.id, user.name, &user.age) == 3) {
+        if (strcmp(user.id ,idToUpdate) == 0) {
+            found = 1;
+            printf("Enter New Name: ");
+    		scanf(" %[^\n]s", user.name);
+    		 while (!isValidName(user.name)) {
+                printf("Invalid Name. Please try again.\nEnter New Name: ");
+                scanf(" %[^\n]s", user.name);
+            } 
+    		printf("Enter Age: ");
+    		scanf("%d", &user.age);
+    		while (!isValidAge(user.age)) {
+                printf("Invalid Age. Please try again.\nEnter New Age: ");
+                scanf("%d", &user.age);
+            }
+
+        }
+        fprintf(tempFile, "%s %s %d\n", user.id, user.name, user.age);
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    if (found) {
+        remove(FILENAME);
+        rename("temp.txt", FILENAME);
+        printf("User modified successfully!\n");
+    } else {
+        remove("temp.txt");
+        printf("User with ID %s not found.\n", idToUpdate);
+    }
+}
+
+void removeUser() {
+    char idToDelete[15], found = 0;
+    User user;
+    FILE *file = fopen(FILENAME, "r");
+    FILE *tempFile = fopen("temp.txt", "w");
+
+    if (file == NULL || tempFile == NULL) {
+        printf("Could not Open file.\n");
+        return;
+    }
+
+    printf("Enter User ID to remove: ");
+    scanf("%s", idToDelete);
+
+while (fscanf(file, "%s %s %d", user.id, user.name, &user.age) == 3) {
+        if (strcmp(user.id,idToDelete)==0) {
+            found = 1;
+        } else {
+            fprintf(tempFile, "%s %s %d\n", user.id, user.name, user.age);
+        }
+    }
+
+    fclose(file);
+    fclose(tempFile);
+
+    if (found) {
+        remove(FILENAME);
+        rename("temp.txt", FILENAME);
+        printf("User removed successfully!\n");
+    } else {
+        remove("temp.txt");
+        printf("User with ID %d not found.\n", idToDelete);
+    }
+}
+
+void initializeFile() {

```

Comment:
Add proper function prototypes before their usage to avoid implicit declaration for(checkDuplicate, isValidName, isValidAge, isdigit)